tests/unit/lib/secrets/vault_csi_tests.py

```diff
@@ -0,0 +1,271 @@
+import datetime
+import json
+import shutil
+import string
+import tempfile
+import typing
+import unittest
+
+from pathlib import Path
+from unittest.mock import mock_open
+from unittest.mock import patch
+
+import gevent
+import pytest
+import typing_extensions
+
+from baseplate.lib.secrets import SecretNotFoundError
+from baseplate.lib.secrets import secrets_store_from_config
+from baseplate.lib.secrets import SecretsStore
+from baseplate.lib.secrets import VaultCSISecretsStore
+
+SecretType: typing_extensions.TypeAlias = typing.Dict[str, any]
+
+
+def write_secrets(secrets_data_path: Path, data: typing.Dict[str, SecretType]) -> None:
+    """Write secrets to the current data directory."""
+    for key, value in data.items():
+        secret_path = secrets_data_path.joinpath(key)
+        secret_path.parent.mkdir(parents=True, exist_ok=True)
+        with open(secret_path, "w") as fp:
+            json.dump(value, fp)
```

Comment:
nit: another option if you prefer, either is OK with me:

```suggestion
        secret_path.write_text(json.dumps(value))
```

Reply:
TIL